DDG4/python/DDSim/Helper/Gun.py

```diff
@@ -114,9 +113,6 @@ def position(self, val):
   def setOptions(self, ddg4Gun):
     """set the starting properties of the DDG4 particle gun"""
     try:
-      ddg4Gun.energy = self.energy  # ddg4Gun.energy actually sets momentum
```

Comment:
I'm starting to doubt this. It seems like `DDG4.GeneratorAction(kernel, "Geant4ParticleGun/" + "Gun").energy` should may have been useful together with some other option.

An alternative to this PR is to revert the confusing behaviour introduced in 91683a6fe8c18ae8d882499d8ebf4db67d7b684d (cc @wdconinc).

Reply:
Let me express my full support for not having a default that throws over a range from 0 to `energy` when the user specifies a single `energy` value.

Reply:
I agree, this is more trouble and a real stumbling block.
What do you think about keeping `energy` as an alias for momentumMax (via `@property`)

Reply:
The reasonable (for high energy physics) approximation of using 'energy' interchangeably for total energy, kinetic energy, and momentum in the context of geant4 simulations really breaks down when in our EIC environment we study e.g. lower limit 150 MeV protons in our tracker (the 150 MeV in this context is momentum).

I personally think that 'energy' is confusing and should not be used without explicit 'total' or 'kinetic'. When momentum min and max setters were introduced, the energy setter was only kept for exact backwards compatibility, not because I'd suggest anyone to use it :-) If there's a way to phase it out, I'd wholeheartedly support that :-)

Reply:
Let's add a "deprecated" warning then, and drop it soon after that.

Reply:
I guess what I wanted to have is a simple way to set mono-energetic particle.
And that used to be the case, but this was eventually (inadvertently) changed https://github.com/AIDASoft/DD4hep/pull/734/files#diff-e6b2229d02a03edb64bc49337cda4bf87560460d23790cc468958479b8e9c72e

I think we need to better define what we want the behavior to be...

Keeping (restoring) the possibility to set the mono-energetic sounds rather more attractive from my point of view.

Reply:
So I restored the behaviour in #949 added some doc text etc. Let me know what you think...